workflow-service/src/main/java/com/redhat/parodos/workflow/execution/controller/WorkFlowController.java

```diff
@@ -113,6 +115,26 @@ public void updateWorkFlowCheckerTaskStatus(@PathVariable UUID workFlowExecution
 				workFlowCheckerTaskRequestDTO.getStatus());
 	}
 
+	@Operation(summary = "Restart a workflow execution with same parameters")
+	@ApiResponses(value = {
+			@ApiResponse(responseCode = "202", description = "Accepted",
+					content = { @Content(mediaType = "application/json",
+							schema = @Schema(implementation = WorkFlowExecutionResponseDTO.class)) }),
+			@ApiResponse(responseCode = "401", description = "Unauthorized", content = @Content),
+			@ApiResponse(responseCode = "403", description = "Forbidden", content = @Content),
+			@ApiResponse(responseCode = "404", description = "Not Found", content = @Content) })
+	@PostMapping("/{workFlowExecutionId}")
+	public ResponseEntity<WorkFlowExecutionResponseDTO> restartWorkFlow(@PathVariable UUID workFlowExecutionId) {
+		WorkReport workReport = workFlowService.restart(workFlowExecutionId);
+		if (workReport.getStatus() == WorkStatus.FAILED) {
```

Comment:
Wouldn't be better to add this logic into workFlowService? And get this covered in test?

Reply:
I don't know, TBH I copy/paste this from the execute, I think this may be a leftover of the synchronous execution.

The only place where the FAILED state is set is in `WorkFlowServiceImpl.execute` :
```
if (validationFailedMsg != null) {
			return new DefaultWorkReport(WorkStatus.FAILED, new WorkContext(), new Throwable(validationFailedMsg));
		}
```
but here maybe we should throws an bad request exception instead, @masayag what do you think?

Reply:
this behavior is aligned with the `execute()`  method, therefore I suggest both be updated.
but this will require a significant amount of changes around tests, so perhaps worth introducing it in a follow-up PR?

Reply:
Agreed! Let's track it a separate ticket for follow-up PR

Reply:
@gabriel-farache +1 for the suggestion to change the service:
```
		String validationFailedMsg = validateWorkflow(workflowName, workFlow);
		if (validationFailedMsg != null) {
			return new DefaultWorkReport(WorkStatus.FAILED, new WorkContext(), new Throwable(validationFailedMsg));
		}
```
should throw "InvalidWorkFlowException" to be caught by the global exception handler.
However, if you look at the implementation of `validateWorkflow` you can see it may produce various errors that each might be better represented by its own case (e.g. resource-not-found,  workflow inner cannot be executed,...).

but I'm good with InvalidWorkFlowException to be translated to a badrequest

Reply:
Please check https://issues.redhat.com/browse/FLPATH-507